lib/queue.rb

```diff
@@ -1,31 +1,60 @@
 class Queue
+  size = 20
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(20)
+    @front = -1
+    @back = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if (@front == 0 && @back ==  size - 1) || (@back == (@front - 1) % size - 1)
+      raise ArgumentError.new("The queue is full!")
+      # queue is empty
+    elsif @front == -1 && @back == -1
+      @front = 0
+      @back = 0
+      # back needs to wrap around
+    elsif @back == size
+      @back = 0
+    else
+      @back = @back + 1
+
+    end
+    @store[@back] = element
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    return nil if @store.empty?
+    
+    raise ArgumentError.new("queue is empty") if @front == -1
+    data = @store[@front]
+    @store[@front] = nil
+
+    if @front == @back
+      @front = -1
+      @back = -1
+    elsif @front == size - 1
+      @front = 0
+    else
+      @front = @front + 1
+    end
+    return data
   end
 
   def front
-    raise NotImplementedError, "Not yet implemented"
+    return @store[@front]
   end
 
   def size
-    raise NotImplementedError, "Not yet implemented"
+    return @store.size
   end
 
   def empty?
-    raise NotImplementedError, "Not yet implemented"
+    @front == @back
   end
 
   def to_s
```

Comment:
Just note that this function will return the elements in the order they are in in the internal array not queue order.  Something to think about.